docker/dockerProxyServer.js

```diff
@@ -0,0 +1,299 @@
+/* eslint-disable no-else-return */
+/* eslint-disable quotes */
+/* eslint-disable no-console */
+'use strict';
+
+const http = require('http');
+const DEFAULT_SERVER_PORT = 3032;
+const cors = require('cors');
+const isDocker = require('is-docker')();
+const version = 1.00;
+
+class DockerProxyServer {
+
+  constructor(port) {
+    this._serverPort = port || DEFAULT_SERVER_PORT;
+    this._requestId = 4000;
+    this._containerId = null;
+    this._server = null;
+    this._instanceLabel = 'dockerProxy' + (new Date()).toISOString();
+    this._originalNetworkPromise = this.getCurrentNetworks();
+  }
+
+  stopServer() {
+    if (this._server) {
+      this._server.close();
+      this._server = null;
+    }
+  }
+
+  startServer() {
+    return this.getCurrentNetworks().then((originalNetworks) => {
+      this._originalNetworks = originalNetworks;
+      return new Promise(resolve => {
+        const express = require('express');
+        const app = express();
+        app.use(cors());
+
+        [
+          { path: '/version', implfn: 'version' },
+          { path: '/isDocker', implfn: 'isDocker' },
+          { path: '/getCurrentContainerId', implfn: 'getCurrentContainerId' },
+          { path: '/getContainers', implfn: 'getContainers' },
+          { path: '/getActiveInterface', implfn: 'getActiveInterface' },
+          { path: '/disconnect/:networkId', implfn: 'disconnectFromNetwork' },
+          { path: '/disconnectFromAllNetworks', implfn: 'disconnectFromAllNetworks' },
+          { path: '/connect/:networkId', implfn: 'connectToNetwork' },
+          { path: '/createNetwork/:networkName', implfn: 'createNetwork' },
+          { path: '/resetNetwork', implfn: 'resetNetwork' },
+          { path: '/inspectCurrentContainer', implfn: 'inspectCurrentContainer' },
+          { path: '/connectToDefaultNetwork', implfn: 'connectToDefaultNetwork' },
+          { path: '/getAllNetworks', implfn: 'getAllNetworks' },
+          { path: '/getCurrentNetworks', implfn: 'getCurrentNetworks' },
+        ].forEach((route) => {
+          app.get(route.path, (req, res, next) => {
+            this[route.implfn](req.params).then(data => res.send(data)).catch(error => {
```

Comment:
@makarandp0 async/await?

Reply:
I will keep this for now, It's more common usage pattern, than async / wait.